### ISSUE TITLE
Support stopping the pressed key from going through

### ISSUE THREAD

**Issue by XamiKami** [obv-mikhail/InputBot#13]
idk if this is realistic to implement (especially cross-plattform), but i'd really need the option to actually "intercept" the keypress in a way that the original key doesn't come through (like for example ahk does it). 
At the moment when i bind to a key, the original keypress still writes letters and such, this should be preventable

**Comment by obv-mikhail** [comment]
@XamiKami Yes, I definitively think this should be implemented, and it should be possible to do so cross-platform. For Windows, there would only need to be a small change made in this function I think - https://github.com/obv-mikhail/InputBot/blob/master/src/windows/mod.rs#L127. Which platform are you using btw?

Do you have any thoughts on how the API could look like? I was thinking of changing the hotkey binding API to be string based (similar to AHK) instead of typed. With string based hotkey binding, a couple things would be easier to implement in the future. But we will lose type-safety in the case..

**Comment by XamiKami** [comment]
Im on Windows and linux, planning a cross-plattform ahk alternative.


What i could imagine would be an api that has some Kind of "event-specificator" api that allows to chain and combine certain conditions and Listen to these. I could try to come up with a mock api tomorrow if it could help.

Id actually recommend against a String based api, because that would make using it harder (no autocomplete)

**Comment by XamiKami** [comment]
i guess no matter the concrete API, a simple boolean-flag "intercept keypress" or something would suffice to add this option. i think if you gave the opportunity to specify key-combinations based on a vec of enum-types (representing keys) that would be a great API, altough that could make it harder to specify things like reacting to key-releases or key-holds or something.

you could make an enum `KeyAction` or something, that has options for `KeyPressed(key: KeyCode)`, `KeyReleased(key: KeyCode)`, `KeyHeld(key: KeyCode, duration: i32)` or something like that, to provide an extensible API for specifying a singular event, and then maybe just accept lists of these to add combined keypresses. (this might be my preferred API, but idk if it fit's what you have in mind for this crate ;D )

**Comment by obv-mikhail** [comment]
@elkowar accepting a list of events is something I would be interested in for sure..

**Comment by SonicZentropy** [comment]
I'm poking into this now (on Windows only thus far) since the crate seems pretty limited without this feature.  Interestingly, while it seems pretty straightforward at first, it also destroys InputBot's ability to handle is_pressed() inside of a bound closure

` AKey.bind(|| {
        if AKey.is_pressed() {
            PKey.press();
        }
}`

The above, for example, will never trigger the P Key, as we've stopped the original AKey press from propagating, which means the call to AKey.is_pressed() fails.  I've not yet found a good way to solve this, but if anyone has ideas, I can keep tinkering.

Edit - I got it working with a kludgy internal cache of pressed keys, but there has to be a cleaner way

[Link to my tinkering repo](https://github.com/SonicZentropy/zenput)
